commit a7d59e121b15fcbaeb60f8a3496843fec2981e51
Author: Simone Gaiarin <simgunz@gmail.com>
Date:   Mon Mar 11 15:28:23 2019 +0000

    REVERT
    openwithdialog: use recursive flag in proxy filter
    
    Summary: Use the flag introduced in Qt 5.10 instead of custom code
    
    Reviewers: dfaure
    
    Reviewed By: dfaure
    
    Subscribers: kde-frameworks-devel
    
    Tags: #frameworks
    
    Differential Revision: https://phabricator.kde.org/D19692

diff --git src/widgets/kopenwithdialog.cpp src/widgets/kopenwithdialog.cpp
index d355216f..c2f4fa8c 100644
--- src/widgets/kopenwithdialog.cpp
+++ src/widgets/kopenwithdialog.cpp
@@ -397,6 +397,14 @@ bool QTreeViewProxyFilter::filterAcceptsRow(int sourceRow, const QModelIndex &pa
         return true;
     }
 
+    //Show the non-leaf node also if the regexp matches one of its children
+    int rows = sourceModel()->rowCount(index);
+    for (int crow = 0; crow < rows; crow++) {
+        if (filterAcceptsRow(crow, index)) {
+            return true;
+        }
+    }
+
     return false;
 }
 
@@ -704,7 +712,6 @@ void KOpenWithDialogPrivate::init(const QString &_text, const QString &_value)
     proxyModel->setSourceModel(appModel);
     proxyModel->setFilterKeyColumn(0);
     proxyModel->setFilterCaseSensitivity(Qt::CaseInsensitive);
-    proxyModel->setRecursiveFilteringEnabled(true);
     view = new KApplicationView(q);
     view->setModels(appModel, proxyModel);
     topLayout->addWidget(view);
